### PR TITLE
Update Flipper UI

### PR DESCRIPTION
**Pull request by ericboehs**
## Summary

- Updated Custom VA Layout to use standard Flipper-UI views so that we can take advantage of gem updates
- Cleaned up logic around authorization in development
  - Presence of `Settings.flipper.github_oauth_key` in `settings.local.yml` will treat app as "production" and require authorization (for future development on flipper feature). Absence of value will treat all developers as authenticated & authorized.
- Added option to Configuration to requirement a verification step before fully enabling a feature

## Related issue(s)

[#91547](https://github.com/department-of-veterans-affairs/va.gov-team/issues/91547)

## Testing done

- [x] *New code is covered by unit tests*
  * Texts fixed for new scenarios

## Screenshots
Unauthenticated Index
image

Unauthenticated Show
image

Authenticated but Unauthorized Index
image

Authenticated and Authorized Show (Index is as expected)
image

Added Config option for requiring second step before fully enabling
image

### CODE REVIEW

File: lib/search/configuration.rb [hunk 1]
```diff
@@ -23,7 +23,7 @@ def mock_enabled?
     end
 
     def base_path
-      flipper_enabled? ? Settings.search.gsa_url : Settings.search.url
+      Settings.search.gsa_url
```
Comment (ericboehs) [inline lib/search/configuration.rb:26]: Remove before merge!

**Comment by LindseySaari** [comment]
NICE! I see a test failure, so I will wait for that. Do we know why we had these custom layouts in the first place?

File: lib/flipper/route_authorization.rb [hunk 2]
```diff
@@ -1,7 +1,7 @@
 # frozen_string_literal: true
 
 module Flipper
-  class AdminUserConstraint
+  class RouteAuthorization
```
Comment (stevenjcumming) [inline lib/flipper/route_authorization.rb:4]: I like the `Constraint` suffix for these types of objects, but I don't feel strongly about it.

**Review by stevenjcumming — APPROVED** [review]

**Review by stevenjcumming — DISMISSED** [review]

File: lib/flipper/README
Comment (rmtolmach) [inline lib/flipper/README:1]: Change to `README.md`

Comment (ryan-mcneil) [reply lib/flipper/README:1]: damn

**Review by rmtolmach — APPROVED** [review]
I pulled the branch down and tested locally. I changed the settings of a feature and it all worked as expected. The pages also load quickly. 

image

**Review by stevenjcumming — APPROVED** [review]

File: lib/flipper/ui/action_patch.rb [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+        path = custom_views_path.join("#{name}.erb") unless custom_views_path.nil?
+
+        # Fall back to default views if custom views haven't been enabled
+        # or if the custom view cannot be found.
+        path = views_path.join("#{name}.erb") if path.nil? || !path.exist?
+
+        raise "Template does not exist: #{path}" unless path.exist?
+
+        # rubocop:disable Security/Eval
+        eval(Erubi::Engine.new(path.read, escape: true).src)
```
Comment (ryan-mcneil) [inline lib/flipper/ui/action_patch.rb:17]: the templates are escaped before eval'd (`escape: true`)

**Pull request merged**